packages/typespec-azure-resource-manager/src/rules/utils.ts

```diff
@@ -28,6 +28,12 @@ export function isTemplatedInterfaceOperation(target: Operation) {
   );
 }
 
+export function isAzureSubNamespace(program: Program, ns: Namespace | undefined): boolean {
```

Comment:
This feels kind of hacky.  Does the core rule have a similar restriction?  If so, can we just borrow its mechanism for checking that this is not library code?

Reply:
yes, this was on core library, took that and deleted the changes on this file.